src/script/api/script_tilelist.cpp

```diff
@@ -83,13 +83,7 @@ ScriptTileList_IndustryAccepting::ScriptTileList_IndustryAccepting(IndustryID in
 	if (i->neutral_station != nullptr && !_settings_game.station.serve_neutral_industries) return;
 
 	/* Check if this industry accepts anything */
-	{
-		bool cargo_accepts = false;
-		for (byte j = 0; j < lengthof(i->accepts_cargo); j++) {
-			if (::IsValidCargoID(i->accepts_cargo[j])) cargo_accepts = true;
-		}
-		if (!cargo_accepts) return;
-	}
+	if (std::all_of(std::begin(i->accepts), std::end(i->accepts), [](const auto &a) { return !::IsValidCargoID(a.cargo); })) return;
```

Comment:
```suggestion
	if (std::none_of(std::begin(i->accepts), std::end(i->accepts), [](const auto &a) { return ::IsValidCargoID(a.cargo); })) return;
```
"There is no valid cargo" vs "All cargos are invalid"